website/content/docs/configuration/keyring/index.mdx

```diff
@@ -0,0 +1,70 @@
+---
+layout: docs
+page_title: keyring Block - Agent Configuration
+description: >-
+  The "keyring" block configures the Nomad servers keyring decryption behavior.
+---
+
+# `keyring` Block
+
+<Placement groups={['keyring']} />
+
+The `keyring` block configures how the Nomad server protects the keyring used
+for encrypting [variables][] and signing [workload identities][]. By default,
+Nomad encrypts the key material with a unique key encryption key (KEK) that is
+not shared between servers, and writes the wrapped key material to disk as
+described in [Key Management][]. Nomad refers to this as `"aead"`. Using the
+`keyring` configuration block, Nomad servers can instead use an external KMS or
+Vault transit encryption. The encrypted key is stored on disk but is now
+protected against offline attacks because the KEK is not present in the file.
+
+All `keyring` blocks require a label for the KMS type. Each external KMS has its
+own configuration options.
+
+```hcl
+keyring [TYPE] {
+  name   = "example"
+  active = true
+}
+```
+
+The default `keyring` configuration is as follows:
+
+```hcl
+keyring "aead" {
+  active = true
+}
+```
+
+## `keyring` Parameters
+
+All `keyring` blocks support the following parameters.
+
+- `name` `(string: "")` - A unique identifier for the `keyring` block, used to
+  disambiguate when there are multiple blocks of the same type.
+
+- `active` `(bool: false)` - Indicates which block to use for encrypting
+  keys. For existing servers, changing which block is `active` only impacts new
+  keys created by a [key rotation][]. Existing keys are encrypted with the
+  previous `active` block, so those blocks should not be removed from the
+  configuration until they have been garbage collected and the keys have been
+  removed from the keystore. In Nomad Community Edition, only a single keyring
+  can be `active` at a time.
+
+## High Availability
+
+<EnterpriseAlert>
+  This functionality only exists in Nomad Enterprise.
+</EnterpriseAlert>
+
+Keyring high availability provides the means to configure multiple active
+`keyring` blocks, in order to have resilience against an outage of an external
+KMS. When there are multiple `keyring` blocks with `active = true`, Nomad
+Enterprise will encrypt each key it creates in all the active KMS providers. On
+startup, Nomad will try each KMS provider in order until it finds a provider
+that can decrypt each key.
```

Comment:
Would it be possible to provide an example of multiple active keyring blocks? I assume adding or removing keyring blocks will require a restart of the server?

Reply:
Good catch on both. Will do.

Reply:
Thanks and if you were to upgrade and migrate from `aeas` to KMS I'm curious what that would look like. Would we need some guidance on perhaps migration to start using this feature?

Reply:
Yeah we designed this to be pretty straightforward to migrate: you update all the servers (with the new config) and then just do a rotation. I'll add a section on that.

Reply:
Amazing, awesome glad to hear that. Great work!

Reply:
I've addressed these items in the most recent commit.